### PR TITLE
[RWDATA-1722] Ability to query ip-to-domains file

### PR DESCRIPTION
**Pull request by bmentges**
Ability to query two files

### CODE REVIEW

File: src/geodata2.erl
Comment (miriampena) [inline src/geodata2.erl:64]: this looks on the right track but I see a bit of repeated code and the funciotn ended up a bit big. Id suggest to move the code from line 47 to 62 to a helper function that can be called twice, each with the right MagellanFile, DomainFile, and Tid. Then from this new function call each of them. I

File: src/geodata2.erl
Comment (miriampena) [inline src/geodata2.erl:62]: As RTB does not use the domain files for now, in my opinion the first file is mandatory and the second one is optional so this new function might need to take it into account. The stop result that is returned in line 62 or 79 have special meaning on the init() function in a gen_server, stoppig the application, and therefore stopping dyno. They mark a strong condition in which there is a severe error and we cant follow on starting the adserver. In this case, if we dont have geodata we cant continue working so the application must stop. If you agree, lets make sure we only load the second file if the first one is loaded ok. But if there is an error the application stops.

File: src/geodata2.erl
Comment (miriampena) [inline src/geodata2.erl:88]: When a init() function gets as return a {stop, Reason} it means that there is a severe error condition and the application must stop. Here MagellanFile could be {stop, ...) which is passed along to new/2 but what should really happen is to just stop the application as a required configuration is missing. A good way to do this is to nest cases like this. 
```
case get_env(geodata2, dbfile) of
        {ok, MagellanFile} ->	     
              case get_env(ip_to_domain, dbfile) of
                   {ok, DomainFile} ->
                         new(MagellanFile, DomainFile);
                   _ ->
                        {stop, {ip_to_domain_dbfile_unspecified}}
               end,
        _ ->	 
             {stop, {geodata2_dbfile_unspecified}}
end.
```

File: src/geodata2.erl
Comment (bmentges) [reply src/geodata2.erl:62]: Understood, and I agree.

File: Makefile
Comment (diegomanuel) [inline Makefile:15]: If `ct` target already calls `compile` (below at line #18) test could be just:
`test: ct`

The way it is now, will run `compile` 2 times when calling `test`.

File: src/geodata2.erl
Comment (diegomanuel) [inline src/geodata2.erl:8]: **DANGER:** The func `is_ipv6_mmdb/0` got removed and it's VITAL to exist!
It's used for boodah and dyno to know if they can lookup for IPv6 or not.

File: src/geodata2.erl
Comment (diegomanuel) [inline src/geodata2.erl:103]: **DANGER:** Those are vital to be kept!!

**Review by diegomanuel — CHANGES_REQUESTED** [review]
**Watch out!** I left some comments that needs to be addressed before merging.

File: test/geodata_SUITE.erl
Comment (miriampena) [inline test/geodata_SUITE.erl:68]: could we add more tests with weird ips, not found ips. and such

File: test/geodata_SUITE.erl
Comment (aagarwalnextroll) [inline test/geodata_SUITE.erl:94]: :nit newline

File: src/geodata2.erl
Comment (bmentges) [reply src/geodata2.erl:8]: Done.

**Review by miriampena — APPROVED** [review]

**Review by diegomanuel — COMMENTED** [review]
LGTM so far, but we need to test-deploy it first at `boodah` and `dyno`!

File: Makefile
Comment (diegomanuel) [inline Makefile:20]: I've created a new target `make test_and_verify` and the idea is to get it to pass all checks in order to have this PR ready to be merged!

**Review by diegomanuel — CHANGES_REQUESTED** [review]
Please, first run `make test_and_verify` and adjust all issues.

**Review by aagarwalnextroll — APPROVED** [review]

**Review by diegomanuel — APPROVED** [review]
Code looks good!
I think it can be merged after **test-deploying** it at `dyno` and `boodah`.
I'll do the `boodah` one as part of: https://adroll.atlassian.net/browse/RTI-8668

**Comment by miriampena** [comment]
tested in dyno perfectly! This is ready on our side for production. How is boodah?

**Comment by diegomanuel** [comment]
> tested in dyno perfectly! This is ready on our side for production. How is boodah?

Same for `boodah`!
Looked healthy: https://github.com/SemanticSugar/boodah/pull/4611
I think `geodata2` is ready to create a new release. 🍻

**Pull request merged**
